### PR TITLE
add list:forAll and list:any

### PR DESCRIPTION
**Pull request by Gryfenfer97**
#23

### CODE REVIEW

File: List.ark [hunk 1]
```diff
@@ -313,3 +313,49 @@
...
+# =end
+# @author https://github.com/Gryfenfer97
+##
+(let list:forAll (fun (a f) {
+    (mut verified true)
+    (mut index 0)
+    (while (and verified (< index (len a))) {
+        (if (not (f (@ a index))) {
+            (set verified false)
+        })
```
Comment (SuperFola) [inline List.ark:334]: you can remove the {} around the set, they are redundant

File: List.ark [hunk 2]
```diff
@@ -313,3 +313,49 @@
...
+# =end
+# @author https://github.com/Gryfenfer97
+##
+(let list:any (fun (a f) {
+    (mut verified false)
+    (mut index 0)
+    (while (and (not verified) (< index (len a))) {
+        (if (f (@ a index)) {
+            (set verified true)
+        })
```
Comment (SuperFola) [inline List.ark:357]: Same thing for the {}

File: tests/list-tests.ark [hunk 3]
```diff
@@ -64,6 +64,11 @@
     (set tests (assert-eq a [1 2 3] "unmodified list" tests))
     (set tests (assert-eq b [4 5 6] "unmodified list" tests))
 
+    (set tests (assert-eq (list:forAll a (fun (e) (< e 4))) true "list:forAll" tests))
+    (set tests (assert-eq (list:forAll a (fun (e) (< e 2))) false "list:forAll" tests))
+    (set tests (assert-eq (list:any a (fun (e) (< e 2))) true "list:any" tests))
+    (set tests (assert-eq (list:any a (fun (e) (> e 8))) false "list:any" tests))
```
Comment (SuperFola) [inline tests/list-tests.ark:70]: Could add two more tests for empty lists

**Review by SuperFola — APPROVED** [review]
